lib/postgrex/extensions/hstore/decoder.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Postgrex.Extensions.Hstore.Decoder do
+  # Types
+  @integer ~r/\A\d+\z/
+  @float ~r/\A\d+\.\d+\z/
+  @null ~r/\ANULL\z/
+
+  @null_value <<255, 255, 255, 255>>
+  # HStore with length of 0
+  def decode(<<0,0,0,0>>) do
+    %{}
+  end
+
+  def decode(<<length::integer-size(32), pairs::binary>> = payload) do
+    decode_payload(%{}, pairs)
+  end
+
+  defp decode_payload(acc, <<>>) do
+    acc
+  end
+
+  # in the case of a NULL value, there won't be a length
+  defp decode_payload(acc, <<key_length::integer-size(32), key::size(key_length)-binary,
+                             @null_value, rest::binary>>) do
```

Comment:
Please use the macros in `BinaryUtils`. `integer-size(32) => int32` `size(length)-binary => binary(length)`.